packages/astro/src/core/dev/update-check.ts

```diff
@@ -0,0 +1,44 @@
+import { fetchPackageJson } from '../../cli/install-package.js';
+import type { AstroPreferences } from '../../preferences/index.js';
+
+export const MAX_PATCH_DISTANCE = 5; // If the patch distance is less than this, don't bother the user
+const CHECK_MS_INTERVAL = 1_036_800_000; // 12 days, give or take
+
+let _latestVersion: string | undefined = undefined;
+
+export async function fetchLatestAstroVersion(
+	preferences: AstroPreferences | undefined
+): Promise<string> {
+	if (_latestVersion) {
+		return _latestVersion;
+	}
+
+	const packageJson = await fetchPackageJson(undefined, 'astro', 'latest');
+	if (packageJson instanceof Error) {
+		throw packageJson;
+	}
+
+	const version = packageJson?.version;
+
+	if (!version) {
+		throw new Error('Failed to fetch latest Astro version');
+	}
+
+	if (preferences) {
+		await preferences.set('_variables.lastUpdateCheck', Date.now(), { reloadServer: false });
```

Comment:
So, I had a choice to make here:
- Use a new file, a new system, etc to store variables like this
- Re-use preferences under a private namespace

I went with the later when I realized adding a new store with different values to preferences would require change a lot of type utils and stuff. I think it's not a problem... You're not supposed to change the config file manually, and the CLI won't show this private namespace.

Reply:
That definitely seems reasonable. We have this same problem with telemetry, too, so probably worth generalizing a pattern at somepoint.